.github/workflows/comment.yml

```diff
@@ -0,0 +1,61 @@
+name: comment
+
+on:
+  workflow_run:
+    workflows: ["build"]
+    types: [completed]
+
+jobs:
+  pr_comment:
+    if: github.event.workflow_run.event == 'pull_request' && github.event.workflow_run.conclusion == 'success'
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/github-script@v6
+        with:
+          script: |
+            const { owner, repo } = context.repo;
+            const run_id = ${{ github.event.workflow_run.id }};
+            const pull_head_sha = '${{github.event.workflow_run.head_sha}}';
+
+            const issue_number = await(async () => {
+              const pulls = await github.rest.pulls.list({ owner, repo });
+              for await (const { data } of github.paginate.iterator(pulls)) {
+                for (const pull of data) {
+                  if (pull.head.sha === pull_head_sha) {
+                    return pull.number;
+                  }
+                }
+              }
+            })();
+            if (issue_number) {
+              core.info(`Using pull request ${issue_number}`);
+            } else {
+              return core.error(`No matching pull request found`);
+            }
+
+            const { data: { artifacts } } = await github.rest.actions.listWorkflowRunArtifacts({ owner, repo, run_id });
+            if (!artifacts.length) {
+              return core.error(`No artifacts found`);
+            }
+            let body = `Download the artifacts for this pull request:\n`;
+            let hidden_windows_artifacts = `\n\n <details><summary>Windows</summary>\n`;
+            for (const art of artifacts) {
+              const art_link = `https://nightly.link/${owner}/${repo}/actions/artifacts/${art.id}.zip`;
+              if (art.name.includes('w64')) {
+                hidden_windows_artifacts += `\n* [${art.name}](${art_link})`;
+              } else {
+                body += `\n* [${art.name}](${art_link})`;
+              }
+            }
+            hidden_windows_artifacts += `\n</details>`;
+            body += hidden_windows_artifacts;
```

Comment:
since all of our artifacts contain "w64" this makes all of them hidden
you can apply the following simplification:
```suggestion
            let body = `Download the artifacts for this pull request:\n\n<details><summary>Windows</summary>`;
            for (const art of artifacts) {
              const art_link = `https://nightly.link/${owner}/${repo}/actions/artifacts/${art.id}.zip`;
              if (art.name.includes('w64')) {
                 body += `\n* [${art.name}](${art_link})`;
              }
            }
            body += `\n</details>`;
```

(you may need to check the code as I don't usually write js)

Reply:
<details><summary> been removed.

Reply:
I thought the summary was good to save vertical space

Reply:
Re-added as your diff